book/listings/scalable_lists/4/main.rs

```diff
@@ -0,0 +1,137 @@
+mod integer_object;
+
+use gtk::{gio, CustomSorter, FilterChange, SorterChange};
+use gtk::{prelude::*, SortListModel};
+use gtk::{
+    Application, ApplicationWindowBuilder, ConstantExpression, Label, ListView, PolicyType,
+    PropertyExpression, ScrolledWindowBuilder, SignalListItemFactory, SingleSelection,
+};
+use integer_object::IntegerObject;
+
+fn main() {
+    // Create a new application
+    let app = Application::new(Some("org.gtk.example"), Default::default());
+    app.connect_activate(build_ui);
+
+    // Run the application
+    app.run();
+}
+
+fn build_ui(application: &Application) {
+    // Create a window
+    let window = ApplicationWindowBuilder::new()
+        .application(application)
+        .title("My GTK App")
+        .default_width(600)
+        .default_height(300)
+        .build();
+
+    let model = gio::ListStore::new(IntegerObject::static_type());
+    for number in 0..=1_000 {
+        let integer_object = IntegerObject::from(number);
+        model.append(&integer_object);
+    }
+
+    let factory = SignalListItemFactory::new();
+    // ANCHOR: factory_setup
+    factory.connect_setup(move |_, list_item| {
+        // Create label
+        let label = Label::new(None);
+        list_item.set_child(Some(&label));
+
+        // Create expression describing `list_item->item->number`
+        let list_item_expression = ConstantExpression::new(list_item);
```

Comment:
Do you need the ConstantExpression here?

Reply:
Not sure.
I ported this nearly 1:1 from the gtk4-demo.